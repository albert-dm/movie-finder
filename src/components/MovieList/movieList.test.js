import React from 'react';
import { shallow } from 'enzyme';
import { MovieList } from './index';

const movies = [
  {
    Title: 'Avengers',
    imdbRating: 9,
    imdbID: 1,
  },
  {
    Title: 'Resident Evil',
    imdbRating: 5.3,
    imdbID: 2,
  },
  {
    Title: 'Matrix',
    imdbRating: 10,
    imdbID: 3,
  },
  {
    Title: 'Titanic',
    imdbRating: 7.2,
    imdbID: 4,
  },
];

describe('Testing MovieList behavior', () => {
  it('Should show search if no movies', () => {
    const component = shallow(<MovieList movies={[]} />);  
    expect(component).toMatchSnapshot();
  });
  it('Should show list if movies', () => {
    const component = shallow(<MovieList movies={movies} />);  
    expect(component).toMatchSnapshot();
  });
});