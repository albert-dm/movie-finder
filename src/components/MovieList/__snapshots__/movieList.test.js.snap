// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieList behavior Should show list if movies 1`] = `
<div
  className="movie-list"
>
  <MovieThumb
    key="1"
    movie={
      Object {
        "Title": "Avengers",
        "imdbID": 1,
        "imdbRating": 9,
      }
    }
    onClick={[Function]}
  />
  <MovieThumb
    key="2"
    movie={
      Object {
        "Title": "Resident Evil",
        "imdbID": 2,
        "imdbRating": 5.3,
      }
    }
    onClick={[Function]}
  />
  <MovieThumb
    key="3"
    movie={
      Object {
        "Title": "Matrix",
        "imdbID": 3,
        "imdbRating": 10,
      }
    }
    onClick={[Function]}
  />
  <MovieThumb
    key="4"
    movie={
      Object {
        "Title": "Titanic",
        "imdbID": 4,
        "imdbRating": 7.2,
      }
    }
    onClick={[Function]}
  />
</div>
`;

exports[`Testing MovieList behavior Should show search if no movies 1`] = `
<div
  className="movie-list"
>
  <div
    className="center"
  >
    <Connect(SearchForm) />
  </div>
</div>
`;
